Library/Homebrew/extend/os/mac/requirements/java_requirement.rb

```diff
@@ -1,13 +1,34 @@
 class JavaRequirement < Requirement
-  cask "java"
+  def initialize(tags)
+    @version = tags.shift if /(\d+\.)+\d/ =~ tags.first
+    @cask = cask
+    super
+  end
 
   env do
     env_java_common
     env_oracle_jdk || env_apple
   end
 
+  # A strict Java 8 requirement (1.8) should prompt the user to install java8
+  def cask
+    if @version.nil? || @version.to_s.end_with?("+") ||
+       @version.to_f >= CURRENT_JAVA.to_f
+      JAVA_CASK_MAP.fetch(CURRENT_JAVA.to_s)
+    else
+      JAVA_CASK_MAP.fetch("1.8")
+    end
+  end
+
   private
 
+  CURRENT_JAVA = 9.0
```

Comment:
@MikeMcQuaid should we retrieve this as `JAVA_CASK_MAP.keys.max` instead of having a separate variable?

Reply:
Avoiding another variable would be ideal 👍